### ISSUE TITLE
label role affects the accesible name computation

### ISSUE THREAD

**Issue by jnurthen** [w3c/accname#43]
_From @jongund on January 14, 2019 17:2_

@accdc The accessible name computation needs to take into account elements with role=label, similar to the accessible name processing for the HTML LABEL element. 

_Copied from original issue: w3c/aria#887_

**Comment by DavidMacDonald** [comment]
I'll be interested in whether the proposal will include the explicit directional label => control as in HTML rather than the label <= control direction of aria-labelledby.

**Comment by accdc** [comment]
Hi David,
Actually this role addition is mainly meant for role parody in the mappings, but won't likely change the way that aria-labelledby works.

E.G aria-labelledby will still need to be able to reference any element with an ID regardless what it's explicit or native role is, so the addition of role=label won't have any impact on how this works.

The questionable impact though, is whether role=label would function like a regular label when it surrounds a native form field, such as:

```
<label>
<input type="checkbox" /> Test
</label>
```

In which case, the AccName probably should do so.

E.G

```
<div role="label">
<input type="checkbox" /> Test
</div>
```

**Comment by jnurthen** [comment]
@devarshipant No. The WAI-ARIA specification cannot require user agents to change native presentation and interaction patterns based on ARIA markup.
